synapse/handlers/groups_local.py

```diff
@@ -383,11 +383,11 @@ def get_publicised_groups_for_user(self, user_id):
 
             defer.returnValue({"groups": result})
         else:
-            result = yield self.transport_client.get_publicised_groups_for_user(
-                get_domain_from_id(user_id), user_id
-            )
+            result = yield self.transport_client.bulk_get_publicised_groups(
+                get_domain_from_id(user_id), [user_id],
+            )["users"][user_id]
```

Comment:
This won't work as you're doing `deferred[..]` rather than `(yield deferred)[..]`. I'd suggest:

```
thign = yield foo
result = thing.get("users", {}).get(user_id)
```